plugins/permission-common/src/PermissionClient.ts

```diff
@@ -91,35 +118,67 @@ export class PermissionClient implements PermissionAuthorizer {
+   * `options` if available.
    *
-   * Authorization requests check that a given Backstage user can perform a protected operation,
-   * potentially for a specific resource (such as a catalog entity). The Backstage identity token
-   * should be included in the `options` if available.
+   * Permissions can be imported from plugins exposing them, such as
+   * `catalogEntityReadPermission`.
    *
-   * Permissions can be imported from plugins exposing them, such as `catalogEntityReadPermission`.
+   * For each query, the response will be either ALLOW or DENY.
    *
-   * The response will be either ALLOW or DENY when either the permission has no resourceType, or a
-   * resourceRef is provided in the request. For permissions with a resourceType, CONDITIONAL may be
-   * returned if no resourceRef is provided in the request. Conditional responses are intended only
-   * for backends which have access to the data source for permissioned resources, so that filters
-   * can be applied when loading collections of resources.
    * @public
    */
   async authorize(
-    queries: EvaluatePermissionRequest[],
-    options?: AuthorizeRequestOptions,
-  ): Promise<EvaluatePermissionResponse[]> {
+    requests: AuthorizePermissionRequest[],
+    options?: EvaluatorRequestOptions,
+  ): Promise<AuthorizePermissionResponse[]> {
     // TODO(permissions): it would be great to provide some kind of typing guarantee that
     // conditional responses will only ever be returned for requests containing a resourceType
     // but no resourceRef. That way clients who aren't prepared to handle filtering according
     // to conditions can be guaranteed that they won't unexpectedly get a CONDITIONAL response.
 
+    return this.makeRequest(requests, authorizeDecisionSchema, options);
+  }
+
+  /**
+   * Fetch the conditional authorization decisions for the given set of
+   * {@link ResourcePermission}s in order to apply the conditions to an upstream
+   * data source.
+   *
+   * @remarks
+   *
+   * For each query, the response will be either ALLOW, DENY, or CONDITIONAL.
+   * Conditional responses are intended only for backends which have access to
+   * the data source for permissioned resources, so that filters can be applied
+   * when loading collections of resources.
+   *
+   * @public
+   */
+  async query(
+    queries: QueryPermissionRequest[],
+    options?: EvaluatorRequestOptions,
+  ): Promise<QueryPermissionResponse[]> {
+    return this.makeRequest(queries, policyDecisionSchema, options);
+  }
+
+  private async makeRequest<TQuery, TResult>(
+    queries: TQuery[],
+    itemSchema: z.ZodSchema<TResult>,
+    options?: AuthorizeRequestOptions,
+  ) {
     if (!this.enabled) {
-      return queries.map(_ => ({ result: AuthorizeResult.ALLOW }));
+      return queries.map(_ => ({ result: AuthorizeResult.ALLOW as const }));
     }
 
-    const request: EvaluatePermissionRequestBatch = {
+    const request = {
```

Comment:
Seems fair to keep this type annotation considering it's what the permission backend is expecting and we don't validate this object before sending the `fetch` request

Reply:
Used `PermissionMessageBatch<TQuery>` instead